packages/flutter_test/test/widget_tester_leaks_test.dart

```diff
@@ -0,0 +1,211 @@
+
+  testWidgets(_test1TrackingOnNoLeaks = 'test1, tracking-on, no leaks', (WidgetTester widgetTester) async {
+    expect(LeakTracking.isStarted, true);
+    expect(LeakTracking.phase.name, _test1TrackingOnNoLeaks);
+    expect(LeakTracking.phase.ignoreLeaks, false);
+    await widgetTester.pumpWidget(Container());
+  });
+
+  testWidgets(_test2TrackingOffLeaks = 'test2, tracking-off, leaks',
+  experimentalLeakTesting: LeakTesting.settings.withIgnoredAll(),
+  (WidgetTester widgetTester) async {
+    expect(LeakTracking.isStarted, true);
+    expect(LeakTracking.phase.name, null);
+    expect(LeakTracking.phase.ignoreLeaks, true);
+    await widgetTester.pumpWidget(StatelessLeakingWidget());
+  });
+
+  testWidgets(_test3TrackingOnLeaks = 'test3, tracking-on, leaks', (WidgetTester widgetTester) async {
+    expect(LeakTracking.isStarted, true);
+    expect(LeakTracking.phase.name, _test3TrackingOnLeaks);
+    expect(LeakTracking.phase.ignoreLeaks, false);
+    await widgetTester.pumpWidget(StatelessLeakingWidget());
+  });
+
+  testWidgets(
+  _test4TrackingOnWithCreationStackTrace = 'test4, tracking-on, with creation stack trace',
+  experimentalLeakTesting: LeakTesting.settings.withCreationStackTrace(),
+    (WidgetTester widgetTester) async {
+      expect(LeakTracking.isStarted, true);
+      expect(LeakTracking.phase.name, _test4TrackingOnWithCreationStackTrace);
+      expect(LeakTracking.phase.ignoreLeaks, false);
+      await widgetTester.pumpWidget(StatelessLeakingWidget());
+    },
+  );
+
+  testWidgets(
+  _test5TrackingOnWithDisposalStackTrace = 'test5, tracking-on, with disposal stack trace',
+  experimentalLeakTesting: LeakTesting.settings.withDisposalStackTrace(),
+    (WidgetTester widgetTester) async {
+      expect(LeakTracking.isStarted, true);
+      expect(LeakTracking.phase.name, _test5TrackingOnWithDisposalStackTrace);
+      expect(LeakTracking.phase.ignoreLeaks, false);
+      await widgetTester.pumpWidget(StatelessLeakingWidget());
+    },
+  );
+
+  // TODO(polina-c): add this test to leak_tracker after merging this code.
+  // It cannot be added here, because it requires flag --enable-vmservice for `flutter test`.
+  // testWidgets(
+  //   _test6TrackingOnWithPath = 'test6, tracking-on, with path',
+  // experimentalLeakTesting: LeakTesting.settings.withRetainingPath(),
+  //   (WidgetTester widgetTester) async {
+  //     expect(LeakTracking.isStarted, true);
+  //     expect(LeakTracking.phase.name, _test6TrackingOnWithPath);
+  //     expect(LeakTracking.phase.ignoreLeaks, false);
+  //     await widgetTester.pumpWidget(StatelessLeakingWidget());
+  //   },
+  // );
+  //
+  // Validation in tear down should be:
+  //
+  // _verifyLeaks(
+  //    reportedLeaks,
+  //    _test6TrackingOnWithPath,
+  //    notDisposed: 1,
+  //    notGCed: 1,
+  //    expectedContextKeys: <LeakType, List<String>>{
+  //      LeakType.notGCed: <String>['path'],
+  //      LeakType.notDisposed: <String>[],
+  //    },
+  // );
```

Comment:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-abandonware